frontend/components/tree/js/tree.service.ts

```diff
@@ -1081,9 +1107,7 @@ export class TreeService {
 
 		}
 
-		setParentNodes.forEach((node) => {
-			this.setSelectionOnParentNodes(node);
-		});
+		this.setSelectionOnParentNodes(node);
```

Comment:
If you decided this function only caters for a single node, not an array, then you should change the function definition and stop passing n an array. The while loop is not necessary anymore.

As it stands the function is a bit odd that it takes in an array, but if the user ends up passing in an array then the function doesn't do the right thing on setSelectionOnParentNodes.